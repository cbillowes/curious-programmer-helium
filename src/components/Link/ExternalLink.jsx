import React, { Component } from "react"
const _ = require("lodash")

class ExternalLink extends Component {
  render() {
    const data = this.props
    const className = data.className ? data.className : ""
    const rel = data.rel ? data.rel : ""

    return (
      <a
        key={`external-link-${_.kebabCase(data.to)}`}
        href={data.to}
        className={className}
        target="_blank"
        rel={`nofollow noopener noreferrer ${rel}`}
      >
        {data.children}
      </a>
    )
  }
}

export default ExternalLink