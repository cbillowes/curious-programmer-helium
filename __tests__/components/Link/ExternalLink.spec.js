import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import ExternalLink from "../../../src/components/Link/ExternalLink"

describe("an external link component", () => {
  it("should render", () => {
    const component = <ExternalLink to="http://www.google.com/">Google</ExternalLink>
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should render contents", () => {
    const component = mount(<ExternalLink to="http://www.google.com/">Google</ExternalLink>)
    expect(component.html()).toEqual(`<a href="http://www.google.com/" class="" target="_blank" rel="nofollow noopener noreferrer ">Google</a>`)
  })

  it("should render link with class name", () => {
    const component = mount(<ExternalLink className="galaxy-swirl" to="http://www.google.com/">Google</ExternalLink>)
    expect(component.html()).toEqual(`<a href="http://www.google.com/" class="galaxy-swirl" target="_blank" rel="nofollow noopener noreferrer ">Google</a>`)
  })

  it("should render link with url rel", () => {
    const component = mount(<ExternalLink rel="license" to="http://www.google.com/">Google</ExternalLink>)
    expect(component.html()).toEqual(`<a href="http://www.google.com/" class="" target="_blank" rel="nofollow noopener noreferrer license">Google</a>`)
  })
})