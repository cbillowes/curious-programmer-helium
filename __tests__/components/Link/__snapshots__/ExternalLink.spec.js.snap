// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`an external link component should render 1`] = `
<a
  className=""
  href="http://www.google.com/"
  rel="nofollow noopener noreferrer "
  target="_blank"
>
  Google
</a>
`;
